### ISSUE TITLE
refactor: xyk config - add assetid and balance

### ISSUE THREAD

**Issue by enthusiastmartin** [galacticcouncil/Basilisk-node#130]
### Description
The make this pallet more usable in future - asset id and balance types should be configurable types in pallet's config.

currently they are set to u32 and u128 respectively as it is defined in primitives crate.

Ideally it would be the best to remove dependency on Basilisk primitives crate ( would depend on #131 )

**Comment by Roznovjak** [comment]
@enthusiastmartin did you have time to took at it? #157

**Comment by enthusiastmartin** [comment]
> 
> 
> @enthusiastmartin did you have time to look at it? #157

I will have a look.

**Comment by enthusiastmartin** [comment]
This is actually much more effort than expected.

To be done properly - math package should be refactored as well - [Math Issue](https://github.com/galacticcouncil/HydraDX-math/issues/13)

Current effort done PR #157 is good start but still requires additional effort to be nice and tidy. 

Marking as Nice To have for now - it is not really necessary atm.

**Comment by enthusiastmartin** [comment]
Not needed
